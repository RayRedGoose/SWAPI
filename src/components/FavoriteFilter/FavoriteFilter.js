import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import './FavoriteFilter.scss'

class FavoriteFilter extends Component {
  constructor(props) {
    super(props)
    this.state = {
      toFavorite: false
    }
  }

  redirect = () => {
    this.setState({toFavorite: true})
  }

  render() {
    return (
      (this.state.toFavorite)
        ? <Redirect to='/favorites' />
        : <button
            className='filter-button '
            onClick={this.redirect}>see favorite characters ({this.props.number})
          </button>
    )
  }
}

export default FavoriteFilter
