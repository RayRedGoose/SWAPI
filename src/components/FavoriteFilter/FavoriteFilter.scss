.filter-button {
  color: #5A2C05;
  cursor: pointer;
  padding: 0.4em 1em;
  background: rgba(255, 255, 255, 0.9);
  border: none;
  font-size: 18px;
  &:hover {
    background: #5A2C05;
    color: #F1AA61;
  }
}
