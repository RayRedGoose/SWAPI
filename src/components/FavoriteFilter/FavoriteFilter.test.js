import React from 'react'
import { shallow } from 'enzyme'
import FavoriteFilter from './FavoriteFilter'

describe('FavoriteFilter', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <FavoriteFilter number={2} />)
  })

  it('should render FavoritesFilter with all data passed in correctly and toFavorite state is false', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render FavoritesFilter with Redirect component if toFavorite state is true', () => {
    wrapper.setState({toFavorite: true})
    expect(wrapper).toMatchSnapshot()
  })

  it('should call redirect with all data passed in correctly', () => {
    const spy = jest.spyOn(wrapper.instance(), 'redirect').mockImplementation(arg => arg)
    wrapper.instance().forceUpdate()

    wrapper.find('button').simulate('click')

    expect(spy).toHaveBeenCalled()
  })

  it('should update state if redirect was called', () => {
    expect(wrapper.state('toFavorite')).toEqual(false)

    wrapper.instance().redirect()

    expect(wrapper.state('toFavorite')).toEqual(true)
  })
})
