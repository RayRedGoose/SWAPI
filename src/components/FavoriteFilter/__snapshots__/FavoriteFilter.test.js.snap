// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteFilter should render FavoritesFilter with Redirect component if toFavorite state is true 1`] = `
<Redirect
  to="/favorites"
/>
`;

exports[`FavoriteFilter should render FavoritesFilter with all data passed in correctly and toFavorite state is false 1`] = `
<button
  className="filter-button "
  onClick={[Function]}
>
  see favorite characters (
  2
  )
</button>
`;
